### PR TITLE
MCR-2236 make annotations@MCRParams, @MCRParam available at runtime

### PR DESCRIPTION
**Pull request by rsteph-de**
[Link to jira](https://mycore.atlassian.net/browse/MCR-2236).

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong on be551c221d68ea4c1cdc68f4430a31e2d82fc66f! 
 Please Check https://github.com/MyCoRe-Travis/test_artifacts/tree/5694-pull_request-1002

File: mycore-restapi/src/main/java/org/mycore/restapi/annotations/MCRParam.java [hunk 1]
```diff
@@ -19,7 +19,10 @@
 package org.mycore.restapi.annotations;
 
 import java.lang.annotation.Annotation;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
 
+@Retention(RetentionPolicy.RUNTIME)
```
Comment (yagee-de) [inline mycore-restapi/src/main/java/org/mycore/restapi/annotations/MCRParam.java:25]: MCRParam should be generated at runtime and not be used to annotate static values

File: mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivates.java [hunk 2]
```diff
@@ -142,6 +142,7 @@ private static void validateDerivateRelation(MCRObjectID mcrId, MCRObjectID derI
 
         },
         tags = MCRRestUtils.TAG_MYCORE_DERIVATE)
+    @MCRParams(values = {@MCRParam(name = MCRMetaDefaultListXMLWriter.PARAM_XMLWRAPPER, value = "derobjects")})
```
Comment (yagee-de) [inline mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivates.java:145]: Should be @XmlElementWrapper instead

**Review by yagee-de — CHANGES_REQUESTED** [review]

File: mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivates.java (hunk 2 above)
Comment (yagee-de) [reply mycore-restapi/src/main/java/org/mycore/restapi/v2/MCRRestDerivates.java:145]: suggest #1003 for this ticket

File: mycore-restapi/src/main/java/org/mycore/restapi/annotations/MCRParam.java (hunk 1 above)
Comment (rsteph-de) [reply mycore-restapi/src/main/java/org/mycore/restapi/annotations/MCRParam.java:25]: Without retention policy the params where not found in code, even if created dynamically

**Pull request closed without merging**